pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -653,12 +672,28 @@ func runCDCMixedVersionCheckpointing(ctx context.Context, t test.Test, c cluster
 		return h.Exec(r, `ALTER TABLE bank.bank SCATTER`)
 	}
 
+	checkIfAffectedBy148620 := func(
+		ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper,
+	) error {
+		for _, v := range []string{"v25.2.0", "v25.2.1", "v25.2.2"} {
+			if h.Context().ToVersion.Equal(clusterupgrade.MustParseVersion(v)) {
+				tester.affectedBy148620 = true
+				return nil
+			}
+		}
+		return nil
+	}
+
 	// Test setup.
 	mvt.OnStartup("start changefeed", tester.createChangeFeed)
 	mvt.OnStartup("create validator", tester.setupValidator)
 	mvt.OnStartup("init workload", tester.initWorkload)
 	mvt.OnStartup("set checkpointing settings", forceCheckpointing)
 
+	// Check before we upgrade to each version whether we're upgrading to
+	// a version that is affeced by the checkpoint fields bug (#148620).
+	mvt.BeforeUpgrade("check if affected by #148620", checkIfAffectedBy148620)
```

Comment:
I think you could just call this before every invocation of `waitAndValidate` and avoid the need for the `BeforeUpgrade` hook all together?

Reply:
Alternatively, maybe we should add a `mvt.RunE` variant which returns an error and the `plan`. Then you can retroactively check `plan.Versions()` to decide whether or not to skip.

Reply:
> Alternatively, maybe we should add a `mvt.RunE` variant which returns an error and the `plan`. Then you can retroactively check `plan.Versions()` to decide whether or not to skip.

I went with this idea